ibc/store/provable_store_test.go

```diff
@@ -0,0 +1,540 @@
+	}{
+		{
+			name:       "create membership proof",
+			membership: true,
+			key:        []byte("key1"),
+			value:      []byte("value1"),
+		},
+		{
+			name:       "create non membership proof",
+			membership: false,
+			key:        []byte("not exists"),
+			value:      nil,
+		},
+	}
+
+	provableStore := newTestProvableStore(t)
+	for _, tc := range tc {
+		t.Run(tc.name, func(t *testing.T) {
+			var proof *ics23.CommitmentProof
+			var err error
+			if tc.membership {
+				proof, err = provableStore.CreateMembershipProof(tc.key, tc.value)
+			} else {
+				proof, err = provableStore.CreateNonMembershipProof(tc.key)
+			}
+			require.NoError(t, err)
+			require.NotNil(t, proof)
+		})
+	}
+}
+
+func TestProvableStore_GetAndProve(t *testing.T) {
+	testCase := []struct {
+		name          string
+		key           []byte
+		expectedValue []byte
+		expectedError error
+	}{
+		{
+			name:          "key exists",
+			key:           []byte("key1"),
+			expectedValue: []byte("value1"),
+			expectedError: nil,
+		},
+		{
+			name:          "key does not exist",
+			key:           []byte("not exists"),
+			expectedValue: nil,
+			expectedError: coreTypes.ErrIBCKeyDoesNotExist("test/not exists"),
+		},
+		{
+			name:          "value is nil",
+			key:           []byte("key2"),
+			expectedValue: nil,
+			expectedError: coreTypes.ErrIBCKeyDoesNotExist("test/key2"),
+		},
+	}
+	provableStore := newTestProvableStore(t)
+	for _, tc := range testCase {
+		t.Run(tc.name, func(t *testing.T) {
+			value, proof, err := provableStore.GetAndProve(tc.key)
+			if tc.expectedError != nil {
+				require.Equal(t, err, tc.expectedError)
+				return
+			}
+			require.Equal(t, value, tc.expectedValue)
+			require.NotNil(t, proof)
+			if tc.expectedValue != nil {
+				require.NotNil(t, proof.GetExist())
+				require.Equal(t, proof.GetExist().Value, tc.expectedValue)
+				require.Equal(t, string(proof.GetExist().Key), "test/"+string(tc.key))
+			} else {
+				require.NotNil(t, proof.GetExclusion())
+				require.Equal(t, string(proof.GetExist().Key), "test/"+string(tc.key))
+			}
+		})
+	}
+}
+
+func TestProvableStore_FlushEntries(t *testing.T) {
+	provableStore := newTestProvableStore(t)
+	kvs := []struct {
+		key   []byte
+		value []byte
+	}{
+		{
+			key:   []byte("testKey1"),
+			value: []byte("testValue1"),
+		},
+		{
+			key:   []byte("testKey2"),
+			value: []byte("testValue2"),
+		},
+		{
+			key:   []byte("testKey3"),
+			value: nil,
+		},
+	}
+	for _, kv := range kvs {
+		if bytes.Equal(kv.value, nil) {
+			err := provableStore.Delete(kv.key)
+			require.NoError(t, err)
+		} else {
+			err := provableStore.Set(kv.key, kv.value)
+			require.NoError(t, err)
+		}
+	}
+	cache := kvstore.NewMemKVStore()
+	require.NoError(t, provableStore.FlushEntries(cache))
+	keys, values, err := cache.GetAll([]byte{}, false)
+	require.NoError(t, err)
+	require.Len(t, keys, 3)
+	for i, key := range keys {
+		require.Equal(t, string(key), "test/1/test/"+string(kvs[i].key))
```

Comment:
Can you move `"test/1/test/"+string(kvs[i].key)` into a var and explain how you got it?